eden/scm/setup.py

```diff
@@ -1669,6 +1669,12 @@ def cythonize(*args, **kwargs):
                 "depends": [versionhashpath],
                 "include_dirs": ["contrib/chg"] + include_dirs,
                 "extra_args": filter(None, cflags + chgcflags + [STDC99, WALL, PIC]),
+                # chg uses libc::unistd/getgroups() to check that chg and the
+                # sl cli have the same permissions (see 49ee17f2bfad).
+                # however, on macOS, getgroups() is limited to NGROUPS_MAX (16) groups by default.
+                # we can work around this by defining _DARWIN_UNLIMITED_GETGROUPS
+                # see https://opensource.apple.com/source/xnu/xnu-3247.1.106/bsd/man/man2/getgroups.2.auto.html
+                "macros": [("_DARWIN_UNLIMITED_GETGROUPS", "1")]
```

Comment:
Should we also write a comment in the calls to `getgroups`? I'm thinking about the situation where a programmer ports the chg client (or server) code to Rust. They'll see a call to `getgroups` and just call `libc::getgroups` without checking that it matches the other half. This would have a chance of reintroducing the bug (because there are no tests to detect it).

In my codebases, I would write `# NOTE[macos-getgroups]: ...` here, and in the `getgroups` calls I would write `# See NOTE[macos-getgroups].` to link to this note. I don't know what the convention is for the Sapling codebase.

Reply:
Honestly the commit message goes much more in-depth than even this comment.  Should I put the commit message in some docs somewhere and reference it here and with the `getgroups` calls?